Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs

```diff
@@ -0,0 +1,143 @@
+                    : ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+            }
+        }
+        private static void OnSpinnerChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is TMSpinner tmSpinner)
+            {
+                tmSpinner._spinnerType = (SpinnerType)newValue;
+            }
+        }
+        private void StartAnimation()
+        {
+            _animationTimer = new Timer(UpdateAnimation, null, TimeSpan.Zero, TimeSpan.FromMilliseconds(animateTimerSeconds));
+        }
+
+        private void UpdateAnimation(object state)
+        {
+            if (_spinnerType == SpinnerType.InDeterminate)
+            {
+                _rotationAngle += 5f;
+            }
+            else if (_spinnerType == SpinnerType.Determinate)
+            {
+                _startAngle += 5f;
+                _sweepAngle += 5f;
+                if (_sweepAngle >= 360f)
+                {
+                    _sweepAngle = 0f;
+                    _startAngle = 0f;
+                }
+            }
+            Device.BeginInvokeOnMainThread(() =>
+            {
+                InvalidateSurface();
+            });
+        }
+        private void DrawCircle(SKPaintSurfaceEventArgs e)
+        {
+            SKImageInfo info = e.Info;
+            SKSurface surface = e.Surface;
+            SKCanvas canvas = surface.Canvas;
+            canvas.Clear();
+
+            float height = Math.Max(info.Height, minHeight);
+            float width = Math.Max(info.Width, minWidth);
+            if (height < minHeight || width < minWidth)
+            {
+                Console.WriteLine("Height or Width < 42");
+            }
+            float diameter = Math.Min(width, height) - 15;
+            float centerX = width / 2;
+            float centerY = height / 2;
+            float radius = diameter / 2;
+
+            SKRect rect = new SKRect(centerX - radius, centerY - radius, centerX + radius, centerY + radius);
+
+            var arcPaint = new SKPaint
+            {
+                Style = SKPaintStyle.Stroke,
+                Color = _spinnerColor.ToSKColor(),
+                StrokeWidth = 15
+            };
+
+            if (_spinnerType == SpinnerType.InDeterminate)
+            {
+                _sweepAngle = 270;
+                canvas.Save();
+                canvas.RotateDegrees(_rotationAngle, centerX, centerY);
+                canvas.DrawArc(rect, _startAngle, _sweepAngle, false, arcPaint);
+                canvas.Restore();
+
+            }
+            else if (_spinnerType == SpinnerType.Determinate)
+            {
+                float startAngle = 270f;
+                float progress = 1f;
+                float endAngle = startAngle + (_sweepAngle * progress);
+                canvas.DrawArc(rect, startAngle, endAngle - startAngle, false, arcPaint);
+            }
+        }
+        #endregion
```

Comment:
Clear the `canvas` after drawn. Check if a clear or flush call available

Reply:
Using canvas.flush()